source/OutfitterPanel.cpp

```diff
@@ -420,52 +424,65 @@ ShopPanel::BuyResult OutfitterPanel::CanBuy(bool onlyOwned) const
+			addError("Only part of your ship's outfit capacity is usable for weapons. "
 				"You cannot install this outfit, because it takes up "
 				+ Format::CargoString(weaponNeeded, "weapon space") + ", and this ship has "
-				+ Format::MassString(weaponSpace) + " free.";
+				+ Format::MassString(weaponSpace) + " free.");
 
 		double engineNeeded = -selectedOutfit->Get("engine capacity");
 		double engineSpace = playerShip->Attributes().Get("engine capacity");
 		if(engineNeeded > engineSpace)
-			return "Only part of your ship's outfit capacity is usable for engines. "
+			addError("Only part of your ship's outfit capacity is usable for engines. "
 				"You cannot install this outfit, because it takes up "
 				+ Format::CargoString(engineNeeded, "engine space") + ", and this ship has "
-				+ Format::MassString(engineSpace) + " free.";
+				+ Format::MassString(engineSpace) + " free.");
 
 		if(selectedOutfit->Category() == "Ammunition")
-			return !playerShip->OutfitCount(selectedOutfit) ?
+			addError(!playerShip->OutfitCount(selectedOutfit) ?
 				"This outfit is ammunition for a weapon. "
 				"You cannot install it without first installing the appropriate weapon."
 				: "You already have the maximum amount of ammunition for this weapon. "
-				"If you want to install more ammunition, you must first install another of these weapons.";
+				"If you want to install more ammunition, you must first install another of these weapons.");
 
 		int mountsNeeded = -selectedOutfit->Get("turret mounts");
 		int mountsFree = playerShip->Attributes().Get("turret mounts");
 		if(mountsNeeded && !mountsFree)
-			return "This weapon is designed to be installed on a turret mount, "
-				"but your ship does not have any unused turret mounts available.";
+			addError("This weapon is designed to be installed on a turret mount, "
+				"but your ship does not have any unused turret mounts available.");
 
 		int gunsNeeded = -selectedOutfit->Get("gun ports");
 		int gunsFree = playerShip->Attributes().Get("gun ports");
 		if(gunsNeeded && !gunsFree)
-			return "This weapon is designed to be installed in a gun port, "
-				"but your ship does not have any unused gun ports available.";
+			addError("This weapon is designed to be installed in a gun port, "
+				"but your ship does not have any unused gun ports available.");
 
 		if(selectedOutfit->Get("installable") < 0.)
-			return "This item is not an outfit that can be installed in a ship.";
+			addError("This item is not an outfit that can be installed in a ship.");
 
 		// For unhandled outfit requirements, show a catch-all error message.
-		return "You cannot install this outfit in your ship, "
+		if(errors.empty())
+		addError("You cannot install this outfit in your ship, "
 			"because it would reduce one of your ship's attributes to a negative amount. "
-			"For example, it may use up more cargo space than you have left.";
+			"For example, it may use up more cargo space than you have left.");
```

Comment:
(github won't let me do it in one comment)
```suggestion
				"For example, it may use up more cargo space than you have left.");
```

Reply:
done